backend/errors/httperror_test.go

```diff
@@ -0,0 +1,64 @@
+package errors
+
+import (
+	"bytes"
+	"errors"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+)
+
+func TestHandleHTTPError(t *testing.T) {
+	tests := []struct {
+		testName       string
+		w              *httptest.ResponseRecorder
+		err            error
+		statusCode     int
+		userMessage    string
+		output         string
+		wrappedMessage string
+		res            bool
+	}{
+		{
+			testName:       "Nil",
+			w:              httptest.NewRecorder(),
+			err:            nil,
+			statusCode:     http.StatusOK,
+			userMessage:    "",
+			output:         "",
+			wrappedMessage: "",
+			res:            false,
+		},
+		{
+			testName:       "Server Error 500",
+			w:              httptest.NewRecorder(),
+			err:            errors.New("foo"),
+			statusCode:     http.StatusInternalServerError,
+			userMessage:    "bar",
+			output:         "Error: foo",
+			wrappedMessage: "{\"error\":\"bar\"}",
+			res:            true,
+		},
+	}
+	for _, test := range tests {
+		currentWriter := log.Writer()
+		var buf bytes.Buffer
+		log.SetOutput(&buf)
```

Comment:
I like how you checked the output of the log message by directing it to the buffer! :100:

Reply:
Thanks, a nice trick I picked up in my Nodejs days.